### PR TITLE
Add process for the recommended rules

### PR DESCRIPTION
**Pull request by laurentlb**
Based on the document https://docs.google.com/document/d/1oYQ-cqmqrpVE02rphobn4F_Q-lqvch4IiUlqEy9q2Fs/edit

### CODE REVIEW

File: recommended-rules.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+---
+
+# Recommended Rules
+
+In the documentation, we provide a list of
+[recommended rules](https://docs.bazel.build/versions/master/rules.html).
+
+This is a set of high quality rules, which will provide a good experience to our
+users. We make a distinction between the supported rules, and the hundreds of
+rules you can find on the Internet.
```
Comment (jin) [inline recommended-rules.md]: Can we elaborate on the distinctions here? e.g. are supported rules also recommended by default? Are bazelbuild/rules_* supported / recommended by default?

File: recommended-rules.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+## Nomination
+
+If a rule meets the requirements below, a rule maintainer can nominate it to be
```
Comment (jin) [inline recommended-rules.md]: rule or ruleset? rule means a single foo_binary() rule to me, whereas ruleset means foo_binary, foo_library, etc.

File: recommended-rules.md (hunk 1 above)
Comment (laurentlb) [reply recommended-rules.md]: There's no special case for rules under `bazelbuild/rules_`. Hopefully most of them will qualify.

File: recommended-rules.md (hunk 2 above)
Comment (laurentlb) [reply recommended-rules.md]: Updated to ruleset

File: recommended-rules.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+If a ruleset meets the requirements below, a rule maintainer can nominate it
+to be part of the _recommended rules_ by filing a
+[GitHub issue](https://github.com/bazelbuild/bazel/).
+
+After a review by the [Bazel core team](https://bazel.build/governance.html), it
+will be recommended on the Bazel website.
+
+## Demotion
+
+If a ruleset stops meeting the requirements (e.g. the code is abandoned, not
```
Comment (dslomov) [inline recommended-rules.md]: Some of the requirements below are vague (I agree with them, but it is difficult to objectively quantify; e.g. what level of test coverage is sufficient?).

I suggest establishing more process, how about smth like this:
- if there is a concern that a particular rule set is not meeting the requirements, a GitHub issue should be filed
- rule maintainers will be contacted and need to respond in X weeks (X=1?)
- Based on the outcome, Bazel core team might make a decision to demote the rule set

File: recommended-rules.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+## Nomination
+
+If a ruleset meets the requirements below, a rule maintainer can nominate it
+to be part of the _recommended rules_ by filing a
+[GitHub issue](https://github.com/bazelbuild/bazel/).
+
+After a review by the [Bazel core team](https://bazel.build/governance.html), it
+will be recommended on the Bazel website.
+
+## Demotion
```
Comment (dslomov) [inline recommended-rules.md]: Suggestion: move this part to the end (good stuff first, bad stuff later)

File: recommended-rules.md (hunk 3 above)
Comment (laurentlb) [reply recommended-rules.md]: Done

**Comment by laurentlb** [comment]
PTAL

**Comment by dslomov** [comment]
Thank you!

**Pull request merged**